### PR TITLE
Add RepoSync e2e test for log level override

### PR DESCRIPTION
**Pull request by victorpras**
Add RepoSync e2e test for log level override that validate overriding one and all containers and clearing all override value

### CODE REVIEW

File: e2e/testcases/override_log_level_test.go [hunk 1]
```diff
@@ -106,3 +109,108 @@ func TestOverrideLogLevel(t *testing.T) {
...
+func TestOverrideRepoSyncLogLevel(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.OverrideAPI, ntopts.NamespaceRepo(frontendNamespace, configsync.RepoSyncName))
+	frontendReconcilerNN := core.NsReconcilerObjectKey(frontendNamespace, configsync.RepoSyncName)
+	frontendNN := nomostest.RepoSyncNN(frontendNamespace, configsync.RepoSyncName)
+	repoSyncFrontend := nomostest.RepoSyncObjectV1Beta1FromNonRootRepo(nt, frontendNN)
+
+	// Verify ns-reconciler-frontend uses the default log level
+	nsReconcilerFrontendDeployment := &appsv1.Deployment{}
+	err := nt.Validate(frontendReconcilerNN.Name, frontendReconcilerNN.Namespace, nsReconcilerFrontendDeployment,
+		testpredicates.DeploymentContainerArgsContains(reconcilermanager.Reconciler, "-v=0"),
```
Comment (tiffanny29631) [inline e2e/testcases/override_log_level_test.go:122]: Checking the log level as desired can be consolidated to a helper function.

Ohterwise lgtm.

Comment (victorpras) [reply e2e/testcases/override_log_level_test.go:122]: Currently the function `DeploymentContainerArgsContains` is used to validate the container args (where log level is configured). What kind of function do you have in mind?

Comment (sdowell) [reply e2e/testcases/override_log_level_test.go:122]: The use of predicates here is fine/readable for a test case IMO

**Review by sdowell — APPROVED** [review]

**Comment by victorpras** [comment]
@sdowell any idea why this PR hasn't been merged yet? Does it need LGTM label?

**Pull request merged**